test/writeEarlyHintsLinks.test.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const eh = require('../index')
+const { Client } = require('undici')
+
+test('Should not add Early Hints', async (t) => {
+  t.plan(1)
+  const payload = { hello: 'world' }
+  const infos = []
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.register(eh)
+  fastify.get('/', (request, reply) => {
+    return payload
+  })
+  await fastify.listen({ port: 0 })
+
+  const client = new Client(`http://localhost:${fastify.server.address().port}`, {
+    keepAliveTimeout: 10,
+    keepAliveMaxTimeout: 10
+  })
+  t.teardown(client.close.bind(client))
+
+  const { body } = await client.request({
+    method: 'GET',
+    path: '/',
+    onInfo: (x) => { infos.push(x) }
+  })
+  await body.dump()
+  t.equal(infos.length, 0)
+})
+
+test('Should add Early Hints headers', async (t) => {
+  t.plan(6)
+  const infos = []
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.register(eh)
+  fastify.get('/', async (request, reply) => {
+    await reply.writeEarlyHintsLinks([
+      'Link: </style.css>; rel=preload; as=style',
+      'Link: </script.js>; rel=preload; as=script'
+    ])
+    return { hello: 'world' }
```

Comment:
I did not see a test that:

```
  fastify.get('/', async (request, reply) => {
    await reply.writeEarlyHintsLinks([
      'Link: </style.css>; rel=preload; as=style',
      'Link: </script.js>; rel=preload; as=script'
    ])
    setTimeout(3 seconds)
    return { hello: 'world' }
```

And check the order of the response's payload:

1- should receive 103 immediately
2- should receive 103 immediately
3- should receive 200 after >= 2 sec

Reply:
`onInfo` already means that it is something before the `actual` response.

Reply:
So moving it before `await body.dump` should be good right?

What I aim, is to intercept Node.js changes/bug if any

Reply:
> should be good right?

Yes, but effect is not obvious here. Since the payload is not large enough, `await client.request` already received the whole payload.

Reply:
Do you think a check like this is useful?

Reply:
I have added a test to ensure the receive order.
It should be good now.
https://github.com/fastify/fastify-early-hints/pull/5/commits/9bca83c404a472fb7632a0df55ba323469ee5a49